modules/composer-service-api/src/main/java/org/opencastproject/composer/api/ComposerService.java

```diff
@@ -77,28 +82,30 @@ Job mux(Track sourceVideoTrack, Track sourceAudioTrack, String profileId) throws
 
   /**
    * Compose two videos into one with an optional watermark.
-   *
-   * @param compositeTrackSize
-   *          The composite track dimension
-   * @param upperTrack
-   *          an optional upper track of the composition
-   * @param lowerTrack
-   *          lower track of the composition
-   * @param watermark
-   *          The optional watermark attachment
-   * @param profileId
-   *          The encoding profile to use
-   * @param background
-   *          The background color
+   * 
+   * @param outputDimension
+   *       The composite track dimension
+   * @param option
+   *        upper track element from mediapackage (optional)
+   * @param lowerLaidOutElement  
+   *        lower track element from mediapackage
+   * @param watermarkOption
+   *        watermark element (optional)
+   * @param identifier
+   *        Encoding profile name
+   * @param outputBackground
+   *        The background color
+   * @param sourceAudioName
+   *        Use audio from only lower or upper track or both, use both when available if omitted
    * @return The receipt for this composite job
    * @throws EncoderException
-   *           if encoding fails
+   *    if encoding fails
    * @throws MediaPackageException
-   *           if the mediapackage is invalid
+   *     if the mediapackage is invalid
    */
-  Job composite(Dimension compositeTrackSize, Option<LaidOutElement<Track>> upperTrack, LaidOutElement<Track> lowerTrack,
-          Option<LaidOutElement<Attachment>> watermark, String profileId, String background) throws EncoderException,
-          MediaPackageException;
+  Job composite(Dimension outputDimension, Option<LaidOutElement<Track>> option,
+        LaidOutElement<Track> lowerLaidOutElement, Option<LaidOutElement<Attachment>> watermarkOption,
+        String identifier, String outputBackground, String sourceAudioName) throws EncoderException, MediaPackageException;
```

Comment:
Can you please either revert _outputBackground_ to _background_ or change the variable name consistently in the interface and its implementation?
The same interface now uses different names for the same formal parameter, e.g. the interface here names it _outputBackground_ but the implementation of the interface in ComposerServiceRemoteImpl names it _background_.  While this is correct, it seems suboptimal the use different names for the same formal parameter.

Reply:
That was an artifact of the patch and I didn't catch it when it was first changed. I will fix it.